packages/assets/src/loader/parsers/textures/utils/createTexture.ts

```diff
@@ -21,6 +21,14 @@ export function createTexture(base: BaseTexture, loader: Loader, url: string)
             Cache.remove(url);
         }
     });
+    // force the destruction of the base texture if the texture is destroyed
+    texture.once('destroyed', () =>
+    {
+        if (!base.destroyed)
+        {
+            base.destroy();
+        }
+    });
```

Comment:
The other changes in the PR are fine, but this is a breaking change.

Users should define if they want to destroy the baseTexture (e.g. `texture.destroy(true)`). We don't handle this automatically, it changes the relationship between Textures and BaseTexture.

Reply:
True, but if the user destroys the `Texture` without the `BaseTexture`, then the asset manager will always return this destroyed `Texture`, and the `BaseTexture` and resource may be never destroyed/released, unless the user calls `baseTexture.destroy()` later (`Assets.unload` cannot destroy it anymore, because the `Texture` is already destroyed). It's should probably not be possible to destroy a `Texture` of the asset manager without destroying the `BaseTexture` in the first place. We could add a warning here when the texture is destroyed without the base. Or we don't destroy the base texture, but remove it from the asset manager. I don't think we would want the asset manager to return destroyed textures. We should probably add a warning in this case as well, because it's very unlikely that the user indented to unload the texture from the asset manager without destroying the resource.

Reply:
A warning seems more appropriate. If someone is using Assets to manage the lifecycle of their assets (recommended), we don't want them calling destroy manually. They should be using unload instead.

Reply:
Removed the destruction of the base texture and replaced it the unloading we do when the base is destroyed and added a warning.